rotkehlchen/tests/unit/decoders/test_velodrome.py

```diff
@@ -0,0 +1,832 @@
+            location=Location.OPTIMISM,
+            event_type=HistoryEventType.TRADE,
+            event_subtype=HistoryEventSubType.RECEIVE,
+            asset=A_OP,
+            balance=Balance(FVal('11.895061655417592619')),
+            location_label=user_address,
+            counterparty=CPT_VELODROME,
+            address=string_to_evm_address('0x2fE304b407c7fAb2a3c10962F14dB751468a4f5b'),  # op_frxeth_pool_address  # noqa: E501
+            notes=F'Receive 11.895061655417592619 OP as the result of a swap in {CPT_VELODROME}',
+        ),
+    ]
+    assert events == expected_events
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('optimism_accounts', [['0xB1D34002ee676516787fd8CDb9C549a415F68aA8']])
+def test_swap_eth_to_token_v1(optimism_accounts, optimism_transaction_decoder):
+    """Check that swapping eth to token in velodrome is properly decoded."""
+    evmhash = deserialize_evm_tx_hash('0xaa6b816c09863eab80f6184ab4112d7fea3a813c194dcebf08b6d4e13df22354')  # noqa: E501
+    user_address = optimism_accounts[0]
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=optimism_transaction_decoder.evm_inquirer,
+        database=optimism_transaction_decoder.database,
+        tx_hash=evmhash,
+    )
+    timestamp = TimestampMS(1695194435000)
+    expected_events = [
+        EvmEvent(
+            tx_hash=evmhash,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.OPTIMISM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(FVal('0.000184626805145159')),
+            location_label=user_address,
+            counterparty=CPT_GAS,
+            notes='Burned 0.000184626805145159 ETH for gas',
+        ), EvmEvent(
+            tx_hash=evmhash,
+            sequence_index=1,
+            timestamp=timestamp,
+            location=Location.OPTIMISM,
+            event_type=HistoryEventType.TRADE,
+            event_subtype=HistoryEventSubType.SPEND,
+            asset=A_ETH,
+            balance=Balance(FVal('0.0000857')),
+            location_label=user_address,
+            counterparty=CPT_VELODROME,
+            address=ROUTER_V1,
+            notes=f'Swap 0.0000857 ETH in {CPT_VELODROME}',
+        ), EvmEvent(
+            tx_hash=evmhash,
+            sequence_index=14,
+            timestamp=timestamp,
+            location=Location.OPTIMISM,
+            event_type=HistoryEventType.TRADE,
+            event_subtype=HistoryEventSubType.RECEIVE,
+            asset=Asset('eip155:10/erc20:0x9485aca5bbBE1667AD97c7fE7C4531a624C8b1ED'),
+            balance=Balance(FVal('0.130694520044655821')),
+            location_label=user_address,
+            counterparty=CPT_VELODROME,
+            address=string_to_evm_address('0x7866C6072B09539fC0FDE82963846b80203d7beb'),
+            notes=F'Receive 0.130694520044655821 agEUR as the result of a swap in {CPT_VELODROME}',
```

Comment:
didn't know capital `F` also works. But please keep it consisten to lowercase across the file. Search if you got any more and adjust them. Won't comment in the PR if I see more.

```suggestion
            notes=f'Receive 0.130694520044655821 agEUR as the result of a swap in {CPT_VELODROME}',
```